ear-production-suite-plugins/plugins/binaural_monitoring/src/bear_data_files.cpp

```diff
@@ -37,114 +30,134 @@ juce::File getCustomDataFileDirectory() {
 
 namespace ear {
 namespace plugin {
-
 DataFileManager::DataFileManager() {
   bearReleaseFiles_.add(
       getBearDataFileDirectory().getChildFile(BEAR_DATA_FILE));
   updateAvailableFiles();
 }
 
+namespace {
+void setFromMetadata(DataFileManager::DataFile& dataFile) {
+  try {
+    auto md = bear::DataFileMetadata::read_from_file(
+        dataFile.fullPath.getFullPathName().toStdString());
+    if (md.has_metadata()) {
+      dataFile.label = md.get_label();
+      dataFile.description = md.get_description();
+      dataFile.isBearRelease = md.is_released();
+    }
+  }
+  catch(std::exception const&) {
+    // What are we (not) handling here?
+  }
+}
+
+void updateFile(juce::File const& file, DataFileManager::FileMap& file_map,
+                bool released) {
+  auto [it, success] = file_map.insert(
+      {file, DataFileManager::DataFile{.filename = file.getFileName(),
+                                       .fullPath = file,
+                                       .label = "",
+                                       .description = "",
+                                       .isBearRelease = released}});
+  if (success) {  // not already added
+    setFromMetadata(it->second);
+  }
+}
+}  // namespace
+
+bool DataFileManager::onlyContainsDefault() const {
+  return customDataFiles.empty() && releasedDataFiles.size() == 1u;
+}
+
 void DataFileManager::updateAvailableFiles() {
-  // Start a new vec and copy existing shared_ptrs of unchanged files 
-  // This retains shared_ptrs unlike clear() and rebuild
-  std::vector<std::shared_ptr<DataFileManager::DataFile>> dfs;
-  // Lookup tfs where we expect to find custom files
-  auto files = getCustomDataFileDirectory().findChildFiles(
-      juce::File::TypesOfFileToFind::findFiles,
-                                    false, "*.tf");
   // add our expected released files
-  for(auto const& bearReleaseFile : bearReleaseFiles_) {
-    if (bearReleaseFile.existsAsFile()) {
-      // note that in win, the released bear dir is the same as the custom
-      // dir, so we might have already found this with findChildFiles
-      files.addIfNotAlreadyThere(bearReleaseFile);
+  for (auto const& file : bearReleaseFiles_) {
```

Comment:
Should we clear out `releasedDataFiles` and `customDataFiles` first?
My original intention was that if the user changed the installed TFs whilst the plugin was running (by deleting/installing) they simply need to reopen the UI for the contents of the dropdown to reflect the files that are now available.